packages/contracts/src/tokens/RewardHolder.sol

```diff
@@ -47,6 +47,8 @@ contract RewardHolder is Initializable, AccessControlUpgradeable, ReentrancyGuar
 
     function __RewardHolder_init_unchained() internal onlyInitializing {
         _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+
+        rewardIndex = 1;
```

Comment:
Just curious why have you moved this initialization here?

Reply:
Otherwise it won't pass the pre-upgrade validation.

Variables initialized in the constructor (or directly when declared) do not pass their values to proxy storage. We need to initialize values in `init` function for this